tests/unit_tests/vecpar/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+
+# make unit tests for multiple algebras
+set( algebras "array" )
+if( DETRAY_EIGEN_PLUGIN )
+    list( APPEND algebras "eigen" )
+endif()
+
+### CPU tests
+find_package(OpenMP QUIET)
+if ((TARGET vecpar::all) AND (TARGET vecpar::omp))
+    foreach(algebra ${algebras})
+       # Unit tests for the selected algebra.
+       detray_add_test( ${algebra}_vecpar_cpu
+               "rk_stepper_vecpar.cpp" "rk_stepper_vecpar.hpp"
+               LINK_LIBRARIES GTest::gtest_main vecmem::cuda detray_tests_common
+               detray::${algebra} vecpar::all OpenMP::OpenMP_CXX)
+
+       target_compile_definitions( detray_test_${algebra}_vecpar_cpu
+               PRIVATE ${algebra}=${algebra} )
+
+       target_compile_options( detray_test_${algebra}_vecpar_cpu PUBLIC
+               $<$<COMPILE_LANGUAGE:CXX>:-std=c++2a>)
+    endforeach()
+endif()
+
+
+### GPU tests can be built only by Clang compiler +nvptx support
+find_package(CUDAToolkit)
+if ((TARGET vecpar::all) AND (TARGET vecpar::cuda))
+    foreach(algebra ${algebras})
+
+        # Unit tests for the selected algebra.
+        detray_add_test(${algebra}_vecpar_gpu
+                "rk_stepper_vecpar.cpp" "rk_stepper_vecpar.hpp"
+                LINK_LIBRARIES GTest::gtest_main vecmem::cuda detray_tests_common
+                detray::${algebra} vecpar::all CUDA::cudart)
+
+        target_compile_definitions( detray_test_${algebra}_vecpar_gpu
+                PRIVATE ${algebra}=${algebra} )
+
+        target_compile_options( detray_test_${algebra}_vecpar_gpu PUBLIC
+                $<$<COMPILE_LANGUAGE:CXX>:-std=c++2a>
+                $<$<COMPILE_LANGUAGE:CXX>:-x cuda --offload-arch=sm_${CMAKE_CUDA_ARCHITECTURES}>)
```

Comment:
Hmm... Is this the design for vecpar then? :thinking:

Note that this forces you into only ever using the same compiler for building C\+\+ and CUDA code in your project. While Clang/LLVM is getting better and better, this is a serious limitation to start out from...

What I've done in the past for "cross compiling" a piece of source code was the following:

```cmake
set_source_files_properties( "rk_stepper_vecpar.cpp" PROPERTIES LANGUAGE CUDA )
if( "${CMAKE_CUDA_COMPILER_ID}" MATCHES "NVIDIA" OR
    "${CMAKE_CUDA_COMPILER_ID}" MATCHES "Clang" )
   set_source_files_properties( "rk_stepper_vecpar.cpp" PROPERTIES
      COMPILE_FLAGS "-x cuda" )
endif()
```

Although do note that I have no idea what even Clang does at this point for C\+\+20 support for CUDA. The CUDA compiler (nvcc) will only support this standard from 11.7, which is still not out if I'm not mistaken...